sharkfuser/include/fusilli/support/asm_emitter.h

```diff
@@ -477,14 +469,11 @@ inline std::string ConvFPropNode::getPermuteYOpsAsm() const {
   std::string suffix = convFPropAttr.getName();
   std::shared_ptr<TensorAttr> yT = convFPropAttr.getY();
 
-  // Emit permute dimensions based on layout.
-  if (yT->isContiguous())
-    oss << getListOfIntOpsAsm(
-        getPreserveContiguousPermuteOrder(yT->getDim().size()), prefix, suffix);
-  else
-    oss << getListOfIntOpsAsm(
-        getContiguousToChannelsLastPermuteOrder(yT->getDim().size()), prefix,
-        suffix);
+  // Emit permute dimensions based on layout (inverse to convert back from
+  // contiguous).
+  oss << getListOfIntOpsAsm(inversePermutation(getLogicalToPhysicalPermuteOrder(
+                                yT->getDim(), yT->getStride())),
```

Comment:
For readability, consider creating a separate utility `getPhysicalToLogicalPermuteOrder` that internally calls `inversePermutation` on the `getLogicalToPhysicalPermuteOrder`...  

(btw these will get swapped after addressing the naming comment..)